src/providers/interpretation_providers/word2vecInterpretationProvider.ts

```diff
@@ -0,0 +1,50 @@
+import w2v from 'word2vec';
+import { promisify } from 'util';
+import { IInterpretationProvider, BusinessInsights, InterpretationParams } from './interpretationProvider';
+
+export interface Word2VecConfig {
+	modelName: string;
+	similarityThreshold: number;
+}
+
+export class Word2VecInterpretationProvider implements IInterpretationProvider {
+	constructor(protected model: any, protected similarityThreshold: number) {}
+
+	static async from(config: Word2VecConfig): Promise<Word2VecInterpretationProvider> {
+		const { modelName, similarityThreshold } = config;
+		const modelPath = `./models/${modelName}`
+		const model = await promisify(w2v.loadModel)(modelPath);
+		return new Word2VecInterpretationProvider(model, similarityThreshold);
+	}
+
+	protected similar(word1: string, word2: string): boolean {
+		return this.model.similarity(word1, word2) > this.similarityThreshold
+	}
+
+	protected isAlcoholic(word: string): boolean {
+		const alcoholicWords = ['beer', 'wine', 'alcohol', 'alcoholic'];
```

Comment:
```suggestion
		const alcoholicWords = ['beer', 'wine', 'alcohol', 'alcoholic', 'cocktail'];
```
From some tests with label and object generation, I've found the word cocktail to also be something that shows up consistently